### DISCUSSION TITLE
Question: What are the main reasons one would choose to migrate to Coyote's async actors programming model?

### DISCUSSION THREAD

**Discussion by jhwj9617** [microsoft/coyote#398] [Q&A] [answered]
I don't think this question was specifically addressed in the docs.
Applying binary rewriting is easy for existing projects.

But where most of our project might use async/await - migrations to a new programming model is a hugely expensive process.
And even starting new projects with this model will introduce inconsistent models of managing concurrency across projects.

What motivations/major benefits are there to consider this migration?

**Comment by akashlal** [comment, marked as answer]
Actors present a different programming model than Tasks, are a more natural fit for some kinds of applications. Actor programs are typically free of locks, and most synchronization is managed by the runtime. If you find yourself designing your application as a bunch of state machines passing messages between themselves (typical for most distributed protocols) then Actors present a more natural way of mapping those state machines to code.

I'd recommend that you use actors only if it simplifies your code. Make this decision early (before writing your code) so that there is no migration overhead. If you already have Tasks code that you're happy with, then no need to migrate.

**Comment by jhwj9617** [comment]
Thanks @akashlal
Good answer
